tests/NationalInstruments.Analyzers.UnitTests/ChainOfMethodsWithLambdasAnalyzerTests.cs

```diff
@@ -642,6 +642,289 @@ private interface ISoftwareContent
+            { 
+                softwareContent.Children(x => false)
+                    .Select(s => s)
+                    .First(s => s.AliasName == ""T"")
+            }
+            .Select(s => s.AliasName)
+            .Where(n => n == ""A"");
+            
+            return newSoftware;
+        }
+
+        private interface ISoftwareContent
+        {
+            IEnumerable<ISoftwareContent> Children(Predicate<int> predicate = null);
+
+            string AliasName { get; }
+        }
+    }
+}
+");
+
+            VerifyDiagnostics(test);
+        }
+
+        [Fact]
+        public void NI1017_WellSplitLambdasJaggedArrayInitializers_NoDiagnostic()
+        {
+            var test = new AutoTestFile(
+@"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        private static int Foo(IEnumerable<int> numbers)
+        {
+            var temp = new int[][]
+            {
+                new[]
+                {
+                numbers.Count(number => number == 1),
+                numbers.Count(number => number == 2),
+                numbers.Count(number => number == 3)
+                },
+
+                new[]
+                {
+                numbers.Count(number => number == 1),
+                numbers.Count(number => number == 2),
+                numbers.Count(number => number == 3)
+                },
+
+                new[]
+                {
+                numbers.Count(number => number == 1),
+                numbers.Count(number => number == 2),
+                numbers.Count(number => number == 3)
+                },
+            };
+            return 42;
+        }
+    }
+}
+");
+
+            VerifyDiagnostics(test);
+        }
+
+        [Fact]
+        public void NI1017_PoorlySplitLambdasJaggedArrayInitializers_EmitsDiagnostic()
+        {
+            var test = new AutoTestFile(
+@"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        private static ISoftwareContent[][] Foo(ISoftwareContent softwareContent)
+        {
+            var temp = new ISoftwareContent[][]
+            {
+                new[]
+                {
+                    <|>softwareContent.Children(x => false).Select(s => s).First(s => s.AliasName == ""T"")
+                },
+                new[]
+                {
+                    <|>softwareContent.Children(x => false).Select(s => s).First(s => s.AliasName == ""U"")
+                }
+            };
+            return temp;
+        }
+
+        private interface ISoftwareContent
+        {
+            IEnumerable<ISoftwareContent> Children(Predicate<int> predicate = null);
+        
+            string AliasName { get; }
+        }
+    }
+}
+",
+GetNI1017Rule(),
+GetNI1017Rule());
+
+            VerifyDiagnostics(test);
+        }
+
```

Comment:
What about a lambda that creates an array with lambas?

Reply:
Thanks for pointing this out! Added test case [here](https://github.com/ni/csharp-styleguide/pull/65/commits/a8c67bafec7f0a615a8cee3a99a3cf4029e4f072).